src/Core/Framework/Adapter/Cache/InvalidatorStorage/RedisInvalidatorStorage.php

```diff
@@ -27,14 +29,63 @@ public function store(array $tags): void
 
     public function loadAndDelete(): array
     {
-        /** @var array{0: list<string>, 1: mixed} $values */
-        $values = $this
-            ->redis
-            ->multi()
-            ->sMembers(self::KEY)
-            ->del(self::KEY)
-            ->exec();
-
-        return $values[0];
+        $tags = $this->loadAndDeleteMultiTransaction();
+
+        if ($tags !== null) {
+            return $tags;
+        }
+
+        return $this->loadAndDeleteSequentialFallback();
+    }
+
+    /**
+     * @return list<string>|null
+     */
+    private function loadAndDeleteMultiTransaction(): ?array
+    {
+        try {
+            /** @var array{0: list<string>, 1: mixed}|false $values */
+            $values = $this
+                ->redis
+                ->multi()
+                ->sMembers(self::KEY)
+                ->del(self::KEY)
+                ->exec();
+
+            if ($values !== false) {
+                return $values[0];
```

Comment:
what if it is empty? Will this return an empty array here?

I would prefer to put the edge case that values is false (foer whatever reason) up in here in the if, and default case (we got a result) in the main execution branch

Reply:
behold! `$values[0]` is `null` if `$values` is an empty array (`[]`)!